private/rdf4cpp/rdf/parser/IStreamQuadIteratorSerdImpl.cpp

```diff
@@ -209,13 +212,13 @@ SerdStatus IStreamQuadIterator::Impl::on_prefix(void *voided_self, SerdNode cons
     if (self->no_parse_prefixes) {
         self->last_error = ParsingError{
                 .error_type = ParsingError::Type::BadSyntax,
-                .line = serd_reader_get_current_line(self->reader.get()),
-                .col = serd_reader_get_current_col(self->reader.get()),
+                .line = serd_reader_get_current_line(self->reader),
+                .col = serd_reader_get_current_col(self->reader),
                 .message = "Encountered prefix while parsing. hint: prefix parsing is currently deactivated. note: position may not be accurate and instead point to the end of the line."};
-    } else {
-        self->state->iri_factory.assign_prefix(node_into_string_view(name), node_into_string_view(uri));
+        return SERD_SUCCESS;
```

Comment:
personally, I preferred here the if-else. If there is an if with an return at the end I expect a short cut. this is none.